### PR TITLE
derive more bundle-headers from pom information by default

### LINKED ISSUE #2447: bnd-maven-plugin: Derive values for some OSGi bundle headers from pom elements
Currently only the following headers are automatically set from pom elements (https://maven.apache.org/pom.html)

1. `Bundle-SymbolicName` derived from `artifactId`
1. `Bundle-Version` derived from `version`
1. `Bundle-Name` derived from `name`

In addition the following headers should be derived as well from the according pom elements automatically:
1. `Bundle-Description` derived from `description`
1. `Bundle-Vendor` derived from `organization/name`
1. `Bundle-License` derived from `licenses/license/name`
1. `Bundle-DocURL` derived from `url`
1. `Bundle-Developers` derived from `developers` (with elements `name`, `email`, `organization`, `roles`)
1. `Bundle-SCM` derived from `scm`

### PR DESCRIPTION
**Pull request by kwin**
This closes #2447

Signed-off-by: Konrad Windszus <kwin@apache.org>

### CODE REVIEW

File: maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java [hunk 1]
```diff
@@ -305,7 +308,108 @@ else if ((wabProperty != null) || hasWablibs) {
...
+			if (builder.getProperty(Constants.BUNDLE_VENDOR) == null) {
+				if (project.getOrganization() != null && StringUtils.isNotBlank(project.getOrganization().getName())) {
+					builder.setProperty(Constants.BUNDLE_VENDOR, project.getOrganization().getName());
+				}
+			}
+			
+			// Set Bundle-License
+			if (builder.getProperty(Constants.BUNDLE_LICENSE) == null) {
+				List<String> licenses = new ArrayList<>();
+				for (License license : project.getLicenses()) {
```
[the remark this answers is not in the document]
Comment (bjhargrave) [reply maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java]: Also, add the `<comment>` tag, if any, to the `description` attribute.

File: maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java [hunk 2]
```diff
@@ -305,7 +308,108 @@ else if ((wabProperty != null) || hasWablibs) {
...
+					licenses.add(licenseValue.toString());
+				}
+				if (licenses.size() > 0) {
+					builder.setProperty(Constants.BUNDLE_LICENSE, StringUtils.join(licenses.iterator(), ","));
+				}
+			}
+			
+			// Set Bundle-SCM
+			if (builder.getProperty(Constants.BUNDLE_SCM) == null) {
+				List<String> scmAttributes = new ArrayList<>();
```
Comment (bjhargrave) [inline maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java]: Again, it seems add to make a list of string which will be joined into a string. Just start with a StringBuilder and add to it.

File: maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java [hunk 3]
```diff
@@ -305,7 +308,108 @@ else if ((wabProperty != null) || hasWablibs) {
...
+					}
+					if (!scmAttributes.isEmpty()) {
+						builder.setProperty(Constants.BUNDLE_SCM, StringUtils.join(scmAttributes.iterator(), ","));
+					}
+				}
+			}
+			
+			// Set Bundle-Developers
+			if (builder.getProperty(Constants.BUNDLE_DEVELOPERS) == null) {
+				List<String> developers = new ArrayList<>();
```
Comment (bjhargrave) [inline maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java]: Same here. Just use a StringBuilder and add to it each loop iteration.

File: maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (kwin) [reply maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java]: the only issue with using a StringBuilder directly is the additional guards you need to check if a separator needs to be inserted. That blows up the code quite a bit, but I will prepare something.

[the remark this answers is not in the document]
Comment (kwin) [reply maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java]: done in https://github.com/bndtools/bnd/pull/2697/commits/215a4367f7a673488bdc97bb37629640cfb804e4.

File: maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java (hunk 2 above)
Comment (kwin) [reply maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java]: done in https://github.com/bndtools/bnd/pull/2697/commits/215a4367f7a673488bdc97bb37629640cfb804e4

File: maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java (hunk 3 above)
Comment (kwin) [reply maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java]: done in https://github.com/bndtools/bnd/pull/2697/commits/215a4367f7a673488bdc97bb37629640cfb804e4

**Comment by bjhargrave** [comment]
Looks good. Can you please squash your commits together and then I can merge. Thanks!

**Comment by kwin** [comment]
@bjhargrave Thanks for the review. I squashed all commits.

**Comment by bjhargrave** [comment]
Thanks for the PR!

**Pull request merged**